### DISCUSSION TITLE
Sign tokens using hydra

### DISCUSSION THREAD

**Discussion by kotyara85** [ory/hydra#3048] [Q&A] [answered]
Hello everyone,
We have our own oauth2 system and we’d like to sign our own tokens using hydra after auth successful in our system which seems to be straight forward. 
the question is there a way to issue secondary tokens (we will use another system) using already issued tokens by hydra?
So it’s more of service kind of tokens for users (cli) etc

Thanks

**Comment by vinckr** [comment]
Can you tell me a bit more about the use case - why double the OAuth2 service? 

This could maybe be achieved much easier when using Ory Oathkeeper
To protect your APIs from unauthorized requests you can use [Ory Oathkeeper](https://www.ory.sh/oathkeeper/docs/#reverse-proxy) - it can read and authorize the tokens created by your upstream OAuth2 service.

**Reply by kotyara85** [reply]
We need service level tokens to be generated with long ttl - several months, primarily for cli and m2m auth. 
we already use opa for authz

**Reply by aeneasr** [reply, marked as answer]
Have you seen https://www.ory.sh/docs/hydra/guides/oauth2-grant-type-jwt-bearer ? I think it is exactly what you need! But please use it with care, it is a very powerful tool and can not be made available to third parties without careful consideration

**Reply by kotyara85** [reply]
That's what the doctor ordered, thanks!
